_posts/2022-mm-dd-crypto-1/2022-mm-dd-crypto1-history-and-basics.markdown

```diff
@@ -0,0 +1,157 @@
+
+> * $\mathcal{P}$ to represent the plain texts
+> * $\mathcal{C}$ for the ciphers
+> * $\mathcal{K}$ for the keys
+> * $\mathcal{E}$ to sum up the encryption rules
+> * $\mathcal{D}$ for the decryption rules
+
+
+And finally, to group them together we'll use brackets to represent them as a tuple:
+> $(\mathcal{P, C, K, E, D})$
+
+By the way, "tuple" tells you that all elements within the brackets always have to be written in that very order (we call this an "ordered list"). Tuples are also always finite.
+
+To be a crypto system not just in name, there's one more rule we need to establish and that is how encryption and decryption work as mathematical functions. Using the above notation, the following must apply to any crypto system:
+> * For every key $K \in \mathcal{K}$, there must be an encryption rule $e_k \in \mathcal{E}$ and a decryption rule $d_k \in \mathcal{D}$.
+> * $e_k$ is defined as a function $e_k: \mathcal{P} \rightarrow \mathcal{C}$
+> * $d_k$ is defined as a function $d_k: \mathcal{C} \rightarrow \mathcal{P}$
+> * For all plaintexts $x \in \mathcal{P}$, $d_k(e_k(x)) = x$ must apply
+
+Let's break this down. What it basically says (in math terms) is that each key in the system must be made in a way that allows you to encrypt and decrypt with it. "Encryption" is a function that creates a cipher text out of a plain text, "decryption" goes the other way around and creates _a_ plain text from a cipher text. And to be useful for its intended purpose, applying a decryption function on a text that has been encrypted using an encryption function from the same crypto system _must_ restore the original plain text.
+The first requirement also introduces an important notation: $K \in \mathcal{K}$ tells us that any key that is held within $\mathcal{K}$ can be represented by the letter $K$. Likewise, plain texts ($P \in \mathcal{P}$) and cipher texts ($C \in \mathcal{C}$) can be generally represented using letters.
+
+The last requirement can be implemented in one of two ways and usually makes the difference between what's called a "symmetrical" or an "asymmetrical" crypto system. Let's look at the definitions for these two.
+
+#### Symmetrical crypto systems
+A symmetrical crypto system is defined as a system where the same key that is used to encrypt the message also decrypts the cipher. This makes them pretty easy to use and fast, at a price. For example, you can't (or at least shouldn't) share the same key with multiple parties, unless you want to share information with all of them. And you'll most likely have to come up with new keys on a regular basis in a sort of arms race between you and Oscar/Eve. Because they will try to get hold of your current key. And if they ever do, a large portion of your communications will be visible for them.
+
```

Comment:
Maybe move the description of asymmetric crypto here, as I'd expect it when we "look at the definitions for these two".

Reply:
Ah, makes sense later while reading.